### ISSUE TITLE
init config parser fails with "snapshot_create no"

### ISSUE THREAD

**Issue by cdpb** [digint/btrbk#360]
In my config I have a section for _internal backups_, _remote backups_ and _external backups_ (external hdd).

All the sections on there own (`btrbk run internal`, ...) work fine, but as soon as I run it all (`btrbk run`) it fails with:

```
ERROR: Subvolume "/btrfs-root/home" and "/btrfs-root/home" will create same snapshot: /btrfs-root/backup/home/home.*
ERROR: Please fix "snapshot_name" configuration options!
```

The _external backups_ just mirror the _internal backups_ to another target like so:

```
# backup to external btrfs device
volume /btrfs-root
  snapshot_create no
  snapshot_preserve_min   all
  target_preserve_min     7d
  target_preserve         48h 7d 4w 6m
  target /mnt/BACKUP/btrfs-backup/

  subvolume home
  group ext ext-home home
        snapshot_dir backup/home

  subvolume root
  group ext ext-root root
        snapshot_dir backup/root
```

The config parser fails because it seems expect unique `snapshot_dir` for every volume. That makes sense to me, but I think it should consider `snapshot_create no` before doing any other processing and just skip every volume with this option.

Thank you for the great software!

Cheers

**Comment by digint** [comment]
Added a fix in: 314c2e74f4f3f8c99eabbc0fa9fba839ed7bc2a7, branch [fix-dup-snap-loc](/digint/btrbk/tree/fix-dup-snap-loc).

Can you please check if it works as expected on your setup?

```
cd /tmp/
wget https://raw.githubusercontent.com/digint/btrbk/fix-dup-snap-loc/btrbk
chmod +x btrbk
sudo ./btrbk run -n
```

Will have to double-check if no other functionality gets broken with this, but should probably be fine.

**Comment by cdpb** [comment]
Works! Thank you

**Comment by digint** [comment]
reopening issue, will close it as soon as it is fully merged and released.

**Comment by digint** [comment]
Did some more tests, works fine.

Merged to master in cee6553a490b80c30d2bcc2979b61ef6a13bfe10

**Comment by digint** [comment]
Included in btrbk-0.31.1, closing bug
